project1/application.py

```diff
@@ -93,24 +93,28 @@ def authenticate():
 
 @app.route('/Search', methods=["GET","POST"])
 def search():
-    Books.query.all()
     
-    isbn = request.form.get("isbn")
-    tittle = request.form.get("tittle")
-    author = request.form.get("author")
-    print("isbn:",isbn)
-    print("tittle is:",tittle)
-    # mem = Books.query.filter_by(isbn = isbn).first()
-    
-    mem = db1.session.query(Books).filter(Books.isbn == isbn).all()
-    print(mem)
-
-
-    # m = Books.query.filter_by(isbn = isbn).first()
-    # s = db1.session.query(Books).filter(or_(Books.isbn==isbn,Books.tittle==tittle,Books.author==author)).all()
-    # print(s)
-    return render_template("Books.html", Books = mem)
-
+    if(request.method == "POST"):
+        book_search = request.form.get("search")
+        
+        if request.form.get("isbnsearch") == "option1":
+            print(book_search +" like  "+"isbnsearch")
+            s = Books.query.filter(Books.isbn.like('%'+ book_search +'%')).all()
```

Comment:
are you able to get partial results also?

Reply:
Yes mam, am able to get all the books displayed on user page. and I have given link to isbn so that it gets redirected to bookpage. For suppose, if I only give "THE" in search box, am getting all the books that has these letters in them

Reply:
ok.One suggestion is for isbn display partial matches with the beginning letters.For the rest it is fine.

Reply:
yes mam, I have implemented this and made a new commit.